<a href="https://colab.research.google.com/github/fedevince/Final-project-Getting-and-cleaning-data-/blob/master/Analisi_Dati_Tesi.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
library("tidyverse")

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.2.1     ✔ readr     2.2.0
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.3     ✔ tibble    3.3.1
✔ lubridate 1.9.5     ✔ tidyr     1.3.2
✔ purrr     1.2.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [ ]:
from google.colab import files
uploaded = files.upload()

In [2]:
library(tidyverse)
cvid_data <- read_csv('DB FEDERICO.csv')

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.2.1     ✔ readr     2.2.0
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.3     ✔ tibble    3.3.1
✔ lubridate 1.9.5     ✔ tidyr     1.3.2
✔ purrr     1.2.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
New names:
• `` -> `...57`
• `` -> `...58`
• `` -> `...59`
Rows: 999 Columns: 59
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (13): ID, BIRTH DATE, TODAY'S DATE, DISEASE ONSET_YEAR, DIAGNOSIS_YEAR, ...
dbl (41): SEX, AGE_YEAR, AGE AT DIAGNOSIS_YEAR, DIAGNOSTIC DELAY_YEAR, TEMPO...
lgl  (5): smB < o > 2%, CD21low < o > 10%, ...57, ...58, ...59

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ S

In [3]:
glimpse(cvid_data)

Rows: 999
Columns: 59
$ ID                                             <chr> "TV_IDCV001", "TV_IDCV0…
$ SEX                                            <dbl> 1, 0, 1, 1, 1, 0, 1, 0,…
$ `BIRTH DATE`                                   <chr> "27/05/1993", "06/01/19…
$ AGE_YEAR                                       <dbl> 32, 36, 72, 37, 53, 51,…
$ `TODAY'S DATE`                                 <chr> "14/05/2026", "14/05/20…
$ `AGE AT DIAGNOSIS_YEAR`                        <dbl> 20, 20, 57, 32, 42, 29,…
$ `DIAGNOSTIC DELAY_YEAR`                        <dbl> 16, 7, 46, 20, 7, 2, 18…
$ `DISEASE ONSET_YEAR`                           <chr> "01/01/1998", "01/01/20…
$ DIAGNOSIS_YEAR                                 <chr> "01/01/2014", "01/01/20…
$ `IDDA AT FU_YEAR`                              <chr> "01/01/2024", "02/01/20…
$ `TEMPO FRA IDDA FU E DIAGNOSI_YEAR`            <dbl> 10, 13, 12, 3, 9, 20, 1…
$ DEATH                                          <dbl> 0, 0, 1, 0, 0, 0, 0, 1,…
$ DEATH_DATE      

# Descriptive Analysis of the cohort


Table of demographics data:

In [24]:
demographics <- cvid_data |> select(ID:DEATH_YEAR)

In [22]:
demographics |>
  summarize(
    num_patients = n_distinct(ID),
    count_sex1 = sum(SEX == 1, na.rm = TRUE),
    prop_sex1 = sum(SEX == 1, na.rm = TRUE) / sum(!is.na(SEX)),
    count_sex0 = sum(SEX == 0, na.rm = TRUE),
    prop_sex0 = sum(SEX == 0, na.rm = TRUE) / sum(!is.na(SEX)),
    median_age_at_diagnosis = median(`AGE AT DIAGNOSIS_YEAR`, na.rm=TRUE),
    iqr_age_at_diagnosis = IQR(`AGE AT DIAGNOSIS_YEAR`, na.rm=TRUE),
    median_age = median(AGE_YEAR, na.rm = TRUE),
    iqr_age = IQR(AGE_YEAR, na.rm = TRUE),
    median_diagnostic_delay = median(`DIAGNOSTIC DELAY_YEAR`, na.rm=TRUE),
    iqr_diagnostic_delay = IQR(`DIAGNOSTIC DELAY_YEAR`, na.rm=TRUE)
    )

num_patients,count_sex1,prop_sex1,count_sex0,prop_sex0,median_age_at_diagnosis,iqr_age_at_diagnosis,median_age,iqr_age,median_diagnostic_delay,iqr_diagnostic_delay
<int>,<int>,<dbl>,<int>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
149,81,0.5472973,67,0.4527027,39,20.75,54,23,5,12.75


* Number of Patients: 149
* Sex 1: 81 patients (54.73%)
* Sex 0: 67 patients (45.27%)
* Median Age at Diagnosis: 39 years (IQR: 20.75)
* Median Age: 54 years (IQR: 23)
* Median Diagnostic Delay: 5 years (IQR: 12.75)



Percentage of Patients who died

In [26]:
demographics |>
  summarize(
    total_patients = n_distinct(ID),
    patients_with_death = sum(DEATH == 1, na.rm = TRUE),
    percentage_death = (sum(DEATH == 1, na.rm = TRUE) / n_distinct(ID))
  )

total_patients,patients_with_death,percentage_death
<int>,<int>,<dbl>
149,12,0.08053691


## AMERATUNGA and IDDA scores

In [27]:
cvid_data |>
  summarize(
    median_idda_score_at_follow_up = median(`IDDA SCORE \nAT FOLLOW UP`, na.rm = TRUE),
    iqr_idda_score_at_follow_up = IQR(`IDDA SCORE \nAT FOLLOW UP`, na.rm = TRUE),
    median_idda_score_at_diagnosis = median(`IDDA SCORE \nAT DIAGNOSIS`, na.rm = TRUE),
    iqr_idda_score_at_diagnosis = IQR(`IDDA SCORE \nAT DIAGNOSIS`, na.rm = TRUE),
    median_ameratunga_score_at_follow_up = median(`AMERATUNGA SCORE AT FOLLOW UP`, na.rm = TRUE),
    iqr_ameratunga_score_at_follow_up = IQR(`AMERATUNGA SCORE AT FOLLOW UP`, na.rm = TRUE),
    median_ameratunga_score_at_diagnosis = median(`AMERATUNGA SCORE AT DIAGNOSIS`, na.rm = TRUE),
    iqr_ameratunga_score_at_diagnosis = IQR(`AMERATUNGA SCORE AT DIAGNOSIS`, na.rm = TRUE)
  )

Warning message:
“There were 2 warnings in `summarize()`.
The first warning was:
ℹ In argument: `median_idda_score_at_follow_up = median(`IDDA SCORE \nAT FOLLOW
  UP`, na.rm = TRUE)`.
Caused by warning in `mean.default()`:
! argument is not numeric or logical: returning NA
ℹ Run `dplyr::last_dplyr_warnings()` to see the 1 remaining warning.”


median_idda_score_at_follow_up,iqr_idda_score_at_follow_up,median_idda_score_at_diagnosis,iqr_idda_score_at_diagnosis,median_ameratunga_score_at_follow_up,iqr_ameratunga_score_at_follow_up,median_ameratunga_score_at_diagnosis,iqr_ameratunga_score_at_diagnosis
<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
NA,36.00015,18,24,14,15,6,9


Scores:
* Median IDDA Score at Follow Up: NA (IQR: 36.00)
* Median IDDA Score at Diagnosis: 18 (IQR: 24)
* Median Ameratunga Score at Follow Up: 14 (IQR: 15)
* Median Ameratunga Score at Diagnosis: 6 (IQR: 9)